src/formats/html.ts

```diff
@@ -75,10 +75,89 @@ export class HtmlImporter extends FormatImporter {
 			return;
 		}
 
+		const fileLookup = new Map<string, { file: PickedFile, tFile: TFile }>;
+		let resolved: Promise<void> | undefined;
+
+		ctx.reportProgress(0, files.length);
 		for (let i = 0; i < files.length; i++) {
 			if (ctx.isCancelled()) return;
 			ctx.reportProgress(i, files.length);
-			await this.processFile(ctx, folder, files[i]);
+			const prevResolved = resolved;
+			resolved = new Promise<void>(resolve => {
```

Comment:
There is a race condition here was the resolved was emitted by a previous file before the next processFile was written, which means this wouldn't really work. Since we're using private APIs anyway we might as well use `MetadataCache.isCacheClean()` instead of keep generating these promises and hooking events.

Reply:
@lishid I updated this to use `MetadataCache.OnCleanCache()`. A quick test using the files in this repo seems to work as expected. Let me know what you think.